crates/ruma-macros/src/events/event_parse.rs

```diff
@@ -230,18 +232,43 @@ pub fn to_kind_variation(ident: &Ident) -> Option<(EventKind, EventKindVariation
 
 pub struct EventEnumEntry {
     pub attrs: Vec<Attribute>,
+    pub aliases: Vec<LitStr>,
     pub ev_type: LitStr,
     pub ev_path: Path,
 }
 
 impl Parse for EventEnumEntry {
     fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
-        let attrs = input.call(Attribute::parse_outer)?;
+        let (ruma_enum_attrs, attrs) = input
+            .call(Attribute::parse_outer)?
+            .into_iter()
+            .partition::<Vec<_>, _>(|attr| attr.path.is_ident("ruma_enum"));
         let ev_type: LitStr = input.parse()?;
         let _: Token![=>] = input.parse()?;
         let ev_path = input.call(Path::parse_mod_style)?;
+        let has_suffix = ev_type.value().ends_with(".*");
+
+        let mut aliases = Vec::with_capacity(ruma_enum_attrs.len());
+        for attr in ruma_enum_attrs {
+            let mut enum_attrs = attr
+                .parse_args_with(Punctuated::<EventEnumAliasAttr, Token![,]>::parse_terminated)?
+                .into_iter()
+                .map(|alias| alias.into_inner())
+                .map(|alias| {
+                    if alias.value().ends_with(".*") == has_suffix {
+                        Ok(alias)
+                    } else {
+                        Err(syn::Error::new_spanned(
+                            &attr,
+                            "aliases should have the same `.*` suffix, or lack thereof, as the main event type",
+                        ))
+                    }
+                })
+                .collect::<syn::Result<_>>()?;
+            aliases.append(&mut enum_attrs);
```

Comment:
Instead of collecting to a `Vec` that's then immediately consumed, I think this should either be a nested loop so we push to `aliases` directly from within the attribute iteration, or a `try_fold` again.